File: Practice/Classic Puzzle - Medium/Mars Lander - Episode 2/Mars Lander - Episode 2.py
import sys
import math

def sign(x):
    if x > 0:
        return 1
    return -1


surface_n = int(input())
land = []
for i in range(surface_n):
    land_x, land_y = [int(j) for j in input().split()]
    land.append([land_x, land_y])
    if i > 0 and land[i][1] == land[i-1][1]:
        landing_y = land[i][1]
        landing_x0 = land[i-1][0]
        landing_x1 = land[i][0]
        landing_x = (landing_x0 + landing_x1) // 2
        landing_length = landing_x1 - landing_x0

while True:
    x, y, h_speed, v_speed, fuel, rotate, power = [int(i) for i in input().split()]

    if abs(landing_x - x) < (landing_length + 100) // 2:
        if abs(h_speed) > 10:
            print(sign(h_speed)*50, 4)
        else:
            print(0, (2, 4)[v_speed < -39])
    else:
        print(20*sign(landing_x-x)*sign(abs(h_speed)-55), 4)

File: Practice/Classic Puzzle - Medium/Mars Lander - Episode 2/Mars Lander - Episode 2.c
#include <stdio.h>

int abs(int x) {
    if (x < 0) {
        return -x;
    }
    return x;
}

int main() {
    int surface_n, last_land_x, last_land_y;
    int landing_y, landing_x, landing_length;
    scanf("%d", &surface_n);
    for (int i = 0; i < surface_n; i++) {
        int land_x;
        int land_y;
        scanf("%d%d", &land_x, &land_y);
        if (i > 0 && land_y == last_land_y) {
            landing_y = land_y;
            landing_x = (land_x + last_land_x) / 2;
            landing_length = abs(land_x - last_land_x);
        }
        last_land_x = land_x;
        last_land_y = land_y;
    }

    while (1) {
        int X, Y, h_speed, v_speed, fuel, rotate, power;
        scanf("%d%d%d%d%d%d%d", &X, &Y, &h_speed, &v_speed, &fuel, &rotate, &power);

        if (abs(landing_x - X) < (landing_length + 100) / 2) {
            if (h_speed > 10) {
                printf("50 4\n");
            } else if (h_speed < -10) {
                printf("-50 4\n");
            } else {
                if (v_speed < -39) {
                    printf("0 4\n");
                } else {
                    printf("0 2\n");
                }
            }
        } else {
            if (landing_x > X) {
                if (h_speed > 55) {
                    printf("20 4\n");
                } else {
                    printf("-20 4\n");
                }
            } else {
                if (h_speed < -55) {
                    printf("-20 4\n");
                } else {
                    printf("20 4\n");
                }
            }
        }
    }

    return 0;
}